pyccs/server.py

```diff
@@ -130,6 +136,33 @@ def receive_response(self, max_size: int, timeout: int = DEFAULT_TIMEOUT_PERIOD)
                                   timeout
                                   )
         return buf
+   
+    #this function is utilized when you might not have a specific size in mind for the message.
+    #when requesting a table for example, you might not know the size of the table in advance.
+    def receive_response_message(self, timeout: int = DEFAULT_TIMEOUT_PERIOD) -> bytearray:
+        #creates a pointer to a c_char and initalizes it to point to NULL
+        buf = POINTER(c_char)()
+        retSize = c_uint()
+        self._lib.CcsRecvResponseMsg(pointer(self._server), 
+                                     pointer(retSize), 
+                                     pointer(buf), 
+                                     timeout
+                                     )
+        
+        if not buf or retSize.value == 0:
+            raise ValueError("Did not receive message/message size is 0")
+        
+        buf_array_type = c_char * retSize.value
+        buf_array = cast(buf, POINTER(buf_array_type)).contents
+
+        # convert back to buffer array 
+        res = bytearray(buf_array[:retSize.value])
+
+        # Free buffer that CcsRecvResponseMsg allocated
+        self._libc.free(buf)
```

Comment:
Can `res` be accessed still after `buf` is freed?

Reply:
yes, this should be tested through the test case, but I will write some more extensive ones to make sure